src/ert/analysis/_es_update.py

```diff
@@ -169,17 +170,38 @@ def _load_observations_and_responses(
 ]:
     try:
         observations_and_responses = ensemble.get_observations_and_responses(
-            [*selected_observations], iens_active_index
+            selected_observations, iens_active_index
         )
+
+        df_observations = pd.DataFrame(
+            {
+                "obs_name": observations_and_responses["obs_name"],
+                "key_index": observations_and_responses["key_index"],
+                "observations": observations_and_responses["observations"],
+                "errors": observations_and_responses["errors"],
+            }
+        )
+
+        df_responses = pd.DataFrame(
+            observations_and_responses["responses"],
+            columns=[f"{i+1}" for i in range(ensemble.ensemble_size)],
+        )
+
+        observations_and_responses_df = pd.concat(
+            [df_observations, df_responses], axis=1
+        ).sort_values(by=["obs_name", "key_index"])
+
     except KeyError as e:
         # Exit early if some observations are pointing to non-existing responses
         raise ErtAnalysisError("No active observations for update step") from e
 
-    responses = observations_and_responses.responses()
-    observations = observations_and_responses.observations()
-    errors = observations_and_responses.errors()
-    observation_keys = observations_and_responses.observation_keys()
-    indexes = observations_and_responses.index()
+    observations = observations_and_responses_df["observations"].to_numpy()
```

Comment:
I think we can get this from `observations_and_responses` and not the dataframe to avoid calling `to_numpy()` again.

Reply:
I think we need to do this back&forth because of the sorting by `obs_name` and `key_index`